extension/observer/hostobserver/extension_test.go

```diff
@@ -184,12 +184,7 @@ func startAndStopObserver(
 	require.NotNil(t, ml.getProcess)
 	require.NotNil(t, ml.collectProcessDetails)
 
-	h := &hostObserver{
-		EndpointsWatcher: observer.EndpointsWatcher{
-			RefreshInterval: 10 * time.Second,
-			Endpointslister: ml,
-		},
-	}
+	h := &hostObserver{EndpointsWatcher: observer.NewEndpointsWatcher(ml, 10*time.Second, zap.NewNop())}
```

Comment:
To make it easier to debug issues in testing, I would suggest using `zaptest.NewLogger` instead of `zap.NewNop`

Reply:
updated